components/sup/src/manager/service/mod.rs

```diff
@@ -936,6 +945,11 @@ impl Service {
                 (false, _) => HealthCheck::Critical,
             }
         };
+        let event_stop = SteadyTime::now();
+        let event_duration = (event_stop - event_start).to_std().unwrap_or_else(|_|
```

Comment:
Why not `unwrap_or_default`?

Additionally, should `event_duration` be an `Option` so we can distinguish between when a hook was actually run or not?

Reply:
Good call on `unwrap_or_default`... for some reason it didn't occur to me that `Duration`'s default would also be 0.

The health check event actually already includes a `has_hook` boolean field that indicates whether the service has actually defined a hook.

Reply:
What would the `event_duration` *mean* in the event `has_hook` is false? It seems like an `Option` eliminates non-sensical states.

Also, I'm not convinced we really *need* `SteadyTime` here. `Instant` already guarantees monotonicity and the stdlib `Instant` says
> useful for tasks such as measuring benchmarks or timing how long an operation takes

Given that, I think we can make this a bit simpler:
```suggestion
        let event_duration = event_start.elapsed();
```
and no need for the previous line.

Reply:
Hrmm, I think I just borrowed the `event_stop - event_start` stuff from similar code elsewhere in the codebase. Using `elapsed` is clearly an improvement. I'll make a note to survey the rest of the code and switch over to `elapsed` where appropriate.

And yes, I should be able to just set the duration to `None` if there was no actual hook to run; that should be a good improvement, as well.